app/assets/javascripts/serviceworker-companion.js

```diff
@@ -1,6 +1,13 @@
 'use strict';
 
-if ('serviceWorker' in navigator) {
+if (
+  'serviceWorker' in navigator &&
+  !(
+    /Safari/i.test(navigator.userAgent) &&
+    /Apple Computer/.test(navigator.vendor)
+  )
```

Comment:
There may be a simpler way to do this. Safari has a `window.safari` property:

In Safari:
<img width="172" alt="Checking Safari for the window.safari property" src="https://user-images.githubusercontent.com/108205/110027885-8c2a8180-7d00-11eb-8969-f139358a4429.png">

In Firefox:
<img width="181" alt="Checking Firefox for the window.safari property" src="https://user-images.githubusercontent.com/108205/110028002-b3814e80-7d00-11eb-9a53-8fa8b35709ef.png">

```suggestion
  !('safari' in window)
```

Reply:
Whoa, very cool! Is there any documentation of this anywhere?

Reply:
Are we sure this check works in Safari mobile as well?

Reply:
It does: I looked at processes running while it was on my iPad. Note dev.to shows twice (once for the loaded page, once for it's service worker). My local forem (joshmacmini.local) only has the web page, which means browser detection is working on iPad and the service worker wasn't loaded!

![CleanShot 2021-03-04 at 15 08 45@2x](https://user-images.githubusercontent.com/37842/110030741-b4fc4800-7cfb-11eb-9275-504b58dbed6f.png)